### PR TITLE
<Calendar /> - close month and year dropdown after selection

### PR DESCRIPTION
**Pull request by bamai**
https://jira.wixpress.com/browse/DSM-1434

### CODE REVIEW

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  You can find it under one of the following URLs:
  * https://preview.wix-style-react.com/pr_6565

File: src/Calendar/DatePickerDropdown/DatePickerDropdown.js [hunk 1]
```diff
@@ -16,36 +16,70 @@ export default class DropdownPicker extends React.Component {
...
 
-  onSelect = data => {
+  state = {
+    open: false,
+  };
+
+  _onSelect = data => {
     const { onChange } = this.props;
 
     if (typeof onChange === 'function') onChange(data);
```
Comment (sivanharel) [inline src/Calendar/DatePickerDropdown/DatePickerDropdown.js:26]: It is better to move it for the setState callback.
React does not guarantee that the state changes are applied immediately.
So it is better to call the onChange method only after the open state was changed to false.

**Review by sivanharel — COMMENTED** [review]
Very nice!
I added some comments

File: src/Calendar/DatePickerDropdown/DatePickerDropdown.js [hunk 2]
```diff
@@ -16,36 +16,70 @@ export default class DropdownPicker extends React.Component {
...
+
+  _onKeyDown = (e, delegateKeyDown) => {
+    delegateKeyDown(e);
+
+    // Close after selection
+    if (
+      e.key === 'Enter' ||
+      e.key === 'Escape' ||
+      e.key === 'Spacebar' ||
+      e.key === ' '
```
[the remark this answers is not in the document]
Comment (sivanharel) [reply src/Calendar/DatePickerDropdown/DatePickerDropdown.js:44]: Actually, this is the right way:
```
    if (e.key === 'Escape') {
      this._close();
      e.preventDefault();
    }

    // prevent TextButton onClick event
    if (open && (e.key === 'Enter' || e.key === 'Spacebar' || e.key === ' ')) {
      console.log('in new if');
      e.preventDefault();
    }
```

File: src/Calendar/DatePickerDropdown/DatePickerDropdown.js [hunk 3]
```diff
@@ -16,36 +16,87 @@ export default class DropdownPicker extends React.Component {
...
+      },
+    );
+  };
+
+  _toggle = () => {
+    this.setState({ open: !this.state.open });
+  };
+
+  _onKeyDown = (e, delegateKeyDown) => {
+    delegateKeyDown(e);
```
Comment (sivanharel) [inline src/Calendar/DatePickerDropdown/DatePickerDropdown.js:41]: You can remove it since you are executing it in the next line :)

**Review by sivanharel — APPROVED** [review]
Nice work!

**Pull request merged**